File: problems/easy/70_climbing_stairs.cpp

// Problem Name: Climbing Stairs
// Difficulty: Easy
// Problem ID: 70
// Tags: Math, Dynamic Programming, Memoization

// Problem Description:
// Write your solution here.

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
    public:
        int climbStairs(int n){
            if(n <= 0){
                return 0;
            }else if(n<4){
                return n;
            }

            int first = 1;
            int second = 1;
            int next = 0;

            for(int i=0; i< n-1; i++){
                next = first+second;
                first = second;
                second = next;
            }
            return next;
        }
};

int main() {
    Solution sl;
    int result = sl.climbStairs(15);
    cout<<result<<endl;
    return 0;
}

File: problems/easy/70_climbing_stairs.py

# Problem Name: Climbing Stairs
# Difficulty: Easy
# Problem ID: 70
# Tags: Math, Dynamic Programming, Memoization

# Problem Description:
# Write your solution here.

# Solution Implementation Here

class Solution:
    def climbStairs(self, n: int) -> int:
        if(n<=0):
            return 0;
        elif(n<4):
            return n;

        first, second, next = 1, 1, 0;

        for i in range(0, n-1):
            next = first + second;
            first =second;
            second = next;

        return next;

# MAIN
sl = Solution();
result = sl.climbStairs(15);
print(result);
